### DISCUSSION TITLE
With the help of library, I have implemented import

### DISCUSSION THREAD

**Discussion by cereschen** [ASDAlexander77/TypeScriptCompiler#91] [Show and tell]
I tried to find a solution to bundle multiple ts files to one ts file, but got nothing

So I decided to implement it myself

The following is the library I just wrote. Welcome to use it

https://www.npmjs.com/package/tsc-bundle

## usage
`npm i tsc-bundle -g`

`tscb use_shared.ts  out.ts`

Then use tsc to run the generated file

example 

```ts
// use_shared.ts
import shared from "./shared";
void function main() {
  console.log(shared.num * 2);
}
```

```ts
// shared.ts
const num = 1
export default {
  num
}
```
## unsupport

```ts
import { num } from "./shared" // comming soon
```

**Comment by ASDAlexander77** [comment]
Interesting, can you show us "real world" example? I mean which u r using to see the "scope" of task

**Reply by cereschen** [reply]
I will package the files that need to be imported in a namespace (I found tsc to have good support for it), and I have a lot of work to do, such as nested imports and handling multiple imports of the same file

Here is the generated code`

```ts
class console {
      static log (str:string) {
        print(str)
      }
    }
namespace shared {
          const num = 1
export default {
  num
}
        }
void function main() {
    console.log(shared.num * 2);
};

```

**Reply by ASDAlexander77** [reply]
BTW, in "class console", I would use "namespace console" or "static class console" to remove useless "hidden" as "static class" is not generating extra code to support "casting" etc

**Reply by cereschen** [reply]
Thanks  for your optimization suggestions. Updated
